common/src/main/java/whocraft/tardis_refined/api/event/TardisEvents.java

```diff
@@ -0,0 +1,43 @@
+package whocraft.tardis_refined.api.event;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.LevelAccessor;
+import whocraft.tardis_refined.common.capability.TardisLevelOperator;
+import whocraft.tardis_refined.common.tardis.IExteriorShell;
+
+public class TardisEvents {
+
+    public static final Event<TakeOff> TAKE_OFF = new Event<>(TakeOff.class, listeners -> (tardisLevelOperator, level, pos) -> Event.result(listeners, takeOff -> takeOff.onTakeOff(tardisLevelOperator, level, pos)));
+
+    public static final Event<Land> LAND = new Event<>(Land.class, listeners -> ((tardisLevelOperator, level, pos) -> {
+        for (Land listener : listeners) {
+            listener.onLand(tardisLevelOperator, level, pos);
+        }
+    }));
+
+    public static final Event<TardisEntry> TARDIS_ENTRY_EVENT = new Event<>(TardisEntry.class, listeners -> (shell, player, externalPos, level, direction) -> {
+        for (TardisEntry listener : listeners) {
+            listener.onEnterTardis(shell, player, externalPos, level, direction);
+        }
+    });
+
+    @FunctionalInterface
+    public interface TardisEntry {
```

Comment:
When this event is fired, we already have an interior dimension registered or created. It would be beneficial for the event to contain the TARDISLevelOperator as they can already fetch the rest of this information from the relevant managers.

Reply:
Fair enough! Will do so